joystick_translator

Hi,

I was trying to get a joystick set up for this project and the project is set up outside of docker. The joystick_translator node would crush with index out of range.

`[ERROR] [1506656598.132534, 1698.131000]: bad callback: <bound method Translator.callback of <__main__.Translator instance at 0x7ff6cb447ea8>>
Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 750, in _invoke_callback
    cb(msg)
  File "/home/febbo/catkin_ws/src/car_demo/car_demo/nodes/joystick_translator", line 43, in callback
    if message.axes[THROTTLE_AXIS] >= 0:
IndexError: tuple index out of range`

After examing with `sudo jstest /dev/input/js0` and the rostopic /joy, I realize my joystick only outputs 3 axes. However, the index `THROTTLE_AXIS` is set to `4` in the source code. I notice someone else joystick's output also has only 3 axes  https://github.com/osrf/car_demo/issues/19.
I simply set `THROTTLE_AXIS` to `1` and it works for my joystick. 
 
I'm not sure if the different number of axes ouptut by our joystick is because we set up the project outside of docker, we are using a different model of joystick or a bug in the code. So I just wanted to open an issue to help people who have similar problems.

--- a/README.md
+++ b/README.md
@@ -23,6 +23,8 @@ This demo has been tested on Ubuntu Xenial (16.04)
 * A joystick
 * A joystick driver which creates links to `/dev/input/js0` or `/dev/input/js1`
 
+This has been tested with the Logitech F710 in Xbox mode. If you have a different joystick you may need to adjust the parameters for the very basic joystick_translator node: https://github.com/osrf/car_demo/blob/master/car_demo/nodes/joystick_translator
+
 # Building
 
 First clone the repo, then run the script `build_demo.bash`.

